### PR TITLE
#279: Ability to skip steps (tools and repositories) during update

### LINKED ISSUE #279: Ability to skip steps such as cloning repositories
As a IDEasy user, I want to be able to skip steps of the IDE project setup so that I can avoid waste of time, bandwith, energy and disc-space e.g. when setting up a project a second time (e.g. for testing changes).

When I create a new project I have to specify a settings git URL. If that settings git contains other git repositories (see #100) that are active, those will be cloned, potentially build and/or imported.
Also there is the variable `[DEVON_]IDE_TOOLS` that configures tools that will be installed automatically.

With this story we should add CLI arguments to optionally skip such things on demand.
My suggestion would be something like this:
* `--skip-repositories`
* `--skip-tools`

Open for discussion would also something more advanced/sophisticated/complex:
* `--skip-repository` / `--skip-repository=foo,bar` / `--skip-repository=*`
* same for `--skip-tool`

WDYT?

### PR DESCRIPTION
**Pull request by salimbouch**
closes #279

### CODE REVIEW

**Review by hohwille — APPROVED** [review]
@salimbouch thanks for your PR. You properly implemented the requested feature/story 👍 
While reading and reviewing the code, I found an earlier bug and also left comments for improvement. Please have a look and update. Then we can merge.

File: cli/src/main/java/com/devonfw/tools/ide/commandlet/AbstractUpdateCommandlet.java [hunk 1]
```diff
@@ -57,7 +62,10 @@ public void run() {
     } finally {
       step.end();
     }
-    updateSoftware();
+
+    if (!skipTools.isTrue()) {
+      updateSoftware();
```
[the remark this answers is not in the document]
Comment (salimbouch) [reply cli/src/main/java/com/devonfw/tools/ide/commandlet/AbstractUpdateCommandlet.java:67]: I agree, done

File: cli/src/main/java/com/devonfw/tools/ide/commandlet/AbstractUpdateCommandlet.java [hunk 2]
```diff
@@ -32,6 +36,7 @@ public AbstractUpdateCommandlet(IdeContext context) {
 
     super(context);
     this.settingsRepo = new StringProperty("", false, "settingsRepository");
+    this.skipTools = new FlagProperty("--skip-tools", false, "skipTools");
```
[the remark this answers is not in the document]
Comment (salimbouch) [reply cli/src/main/java/com/devonfw/tools/ide/commandlet/AbstractUpdateCommandlet.java]: done

File: cli/src/main/java/com/devonfw/tools/ide/commandlet/CreateCommandlet.java [hunk 3]
```diff
@@ -54,7 +61,9 @@ public void run() {
     initializeProject(newProjectPath);
     this.context.setIdeHome(newProjectPath);
     super.run();
-    updateRepositories();
+    if (!this.skipRepositories.isTrue()) {
+      updateRepositories();
+    }
```
[the remark this answers is not in the document]
Comment (salimbouch) [reply cli/src/main/java/com/devonfw/tools/ide/commandlet/CreateCommandlet.java]: done

File: cli/src/main/java/com/devonfw/tools/ide/commandlet/AbstractUpdateCommandlet.java [hunk 4]
```diff
@@ -57,7 +62,10 @@ public void run() {
     } finally {
       step.end();
     }
-    updateSoftware();
+
+    if (!skipTools.isTrue()) {
+      updateSoftware();
+    }
```
[the remark this answers is not in the document]
Comment (salimbouch) [reply cli/src/main/java/com/devonfw/tools/ide/commandlet/AbstractUpdateCommandlet.java]: done

File: cli/src/main/java/com/devonfw/tools/ide/commandlet/AbstractUpdateCommandlet.java [hunk 5]
```diff
@@ -57,7 +62,10 @@ public void run() {
     } finally {
       step.end();
     }
-    updateSoftware();
+
```
[the remark this answers is not in the document]
Comment (salimbouch) [reply cli/src/main/java/com/devonfw/tools/ide/commandlet/AbstractUpdateCommandlet.java:65]: done

[the remark this answers is not in the document]
Comment (hohwille) [reply cli/src/main/java/com/devonfw/tools/ide/commandlet/AbstractUpdateCommandlet.java]: Thank you very much 👍

**Comment by hohwille** [comment]
@salimbouch your rework is awesome but the CI build is red:
```
[ERROR]   CompleteTest.testCompleteEmpty:36 
...
[ERROR]   CompleteTest.testCompleteEmptyNoCtxOptions:53 
Expecting actual:
...
but some elements were not found:
  ["create", "update"]
and others were not expected:
  ["--skip-tools", "--skip-tools"]
```

I am sorry if this broke due to my suggested improvements.
Can you try to also move the `addKeyword(getName())` statement to the super-class as suggested by my review (before the skipTools option)? IMHO that should fix the test.

We could also fix the auto-completion in this regard but that will be more tricky and would better be done in a separate PR.

**Comment by hohwille** [comment]
> @salimbouch your rework is awesome but the CI build is red:
> 
> ```
> [ERROR]   CompleteTest.testCompleteEmpty:36 
> ...
> [ERROR]   CompleteTest.testCompleteEmptyNoCtxOptions:53 
> Expecting actual:
> ...
> but some elements were not found:
>   ["create", "update"]
> and others were not expected:
>   ["--skip-tools", "--skip-tools"]
> ```
> 
> I am sorry if this broke due to my suggested improvements. Can you try to also move the `addKeyword(getName())` statement to the super-class as suggested by my review (before the skipTools option)? IMHO that should fix the test.
> 
> We could also fix the auto-completion in this regard but that will be more tricky and would better be done in a separate PR.

Fixed as constructive review with github web-IDE.

**Pull request merged**